pallets/confidential-asset/src/lib.rs

```diff
@@ -370,33 +450,256 @@ decl_module! {
 
             Ok(())
         }
+
+        /// Resets the `FailedOutgoingBalance` and `IncomingBalance` accumulators for the caller's account.
+        /// If successful, the account owner must use their current balance minus the sum of all unsettled outgoing
+        /// balances as their pending balance.
+        ///
+        /// # Arguments
+        /// * `origin` - contains the secondary key of the caller (i.e who signed the transaction to execute this function).
+        /// * `account_id` - the mercat account ID of the `origin`.
+        ///
+        /// # Errors
+        /// - `BadOrigin` if not signed.
+        ///
+        /// # Weight
+        /// `3_000_000_000`
+        #[weight = 3_000_000_000]
+        pub fn reset_ordering_state(
+            origin,
+            account_id: MercatAccountId,
+        ) {
+            let owner = ensure_signed(origin)?;
+            let owner_did = Context::current_identity_or::<Identity<T>>(&owner)?;
+
+            Self::reset_mercat_pending_state(&owner_did, &account_id);
+
+            // Emit an event and include account's current balance for account owner's information.
+            let current_balance =
+                Self::mercat_account_balance(owner_did, &account_id);
+            Self::deposit_event(RawEvent::ResetConfidentialAccountOrderingState(
+                owner_did,
+                account_id,
+                current_balance
+            ));
+        }
     }
 }
 
 impl<T: Trait> Module<T> {
-    pub fn update_mercat_account_balance(
+    /// Reset `IncomingBalance` and `FailedOutgoingBalance` accumulators, by doing this
+    /// we are no longer keeping track of the old/settled transactions.
+    fn reset_mercat_pending_state(owner_did: &IdentityId, account_id: &MercatAccountId) {
+        if IncomingBalance::contains_key(owner_did, account_id) {
+            IncomingBalance::remove(owner_did, account_id);
+        }
+
+        if FailedOutgoingBalance::contains_key(owner_did, account_id) {
+            FailedOutgoingBalance::remove(owner_did, account_id);
```

Comment:
The `if` checks are not required, you can directly call `remove` in any case without worrying about any perf penalty.

Reply:
Good to know!